node/impl/full/multisig.go

```diff
@@ -0,0 +1,282 @@
+	if actErr != nil {
+		return api.UndefAddressPair, actErr
+	}
+
+	// new actors are created by invoking 'exec' on the init actor with the constructor params
+	execParams := &init_.ExecParams{
+		CodeCID:           builtin.MultisigActorCodeID,
+		ConstructorParams: enc,
+	}
+
+	enc, actErr = actors.SerializeParams(execParams)
+	if actErr != nil {
+		return api.UndefAddressPair, actErr
+	}
+
+	// now we create the message to send this with
+	msg := types.Message{
+		To:       builtin.InitActorAddr,
+		From:     src,
+		Method:   builtin.MethodsInit.Exec,
+		Params:   enc,
+		GasPrice: gp,
+		GasLimit: 1000000,
+		Value:    val,
+	}
+
+	// send the message out to the network
+	smsg, err := a.MpoolAPI.MpoolPushMessage(ctx, &msg)
+	if err != nil {
+		return api.UndefAddressPair, err
+	}
+
+	fmt.Println("send approval in message: ", smsg.Cid())
+
+	// wait for it to get mined into a block
+	wait, err := a.StateAPI.StateWaitMsg(ctx, smsg.Cid())
+	if err != nil {
+		return api.UndefAddressPair, err
+	}
+
+	// check it executed successfully
+	if wait.Receipt.ExitCode != 0 {
+		fmt.Println("actor creation failed!")
+		return api.UndefAddressPair, err
+	}
+
+	// get address of newly created miner
+
+	var execreturn init_.ExecReturn
+	if err := execreturn.UnmarshalCBOR(bytes.NewReader(wait.Receipt.Return)); err != nil {
+		return api.UndefAddressPair, err
+	}
+	fmt.Println("Created new multisig: ", execreturn.IDAddress, execreturn.RobustAddress)
+
+	return api.AddressPair{
+		IDAddress:     execreturn.IDAddress,
+		RobustAddress: execreturn.RobustAddress,
+	}, nil
+}
+
+func (a *MsigAPI) MsigPropose(ctx context.Context, msig address.Address, to address.Address, amt types.BigInt, src address.Address, method uint64, params []byte) (api.MsigProposeOutput, error) {
+
+	if msig == address.Undef {
+		return api.UndefMsigProposalOutput, xerrors.Errorf("must provide a multisig address for proposal")
+	}
+
+	if to == address.Undef {
+		return api.UndefMsigProposalOutput, xerrors.Errorf("must provide a target address for proposal")
+	}
+
+	if !types.BigPos(amt) {
+		return api.UndefMsigProposalOutput, xerrors.Errorf("must provide a positive amount for proposed send")
+	}
+
+	if src == address.Undef {
+		defaddr, err := a.WalletAPI.WalletDefaultAddress(ctx)
+		if err != nil {
+			return api.UndefMsigProposalOutput, err
+		}
+		src = defaddr
+	}
+
+	enc, actErr := actors.SerializeParams(&samsig.ProposeParams{
+		To:     to,
+		Value:  amt,
+		Method: abi.MethodNum(method),
+		Params: params,
+	})
+	if actErr != nil {
+		return api.UndefMsigProposalOutput, actErr
+	}
+
+	msg := &types.Message{
+		To:       msig,
+		From:     src,
+		Value:    types.NewInt(0),
+		Method:   builtin.MethodsMultisig.Propose,
+		Params:   enc,
+		GasLimit: 100000,
+		GasPrice: types.NewInt(1),
+	}
+
+	smsg, err := a.MpoolAPI.MpoolPushMessage(ctx, msg)
+	if err != nil {
+		return api.UndefMsigProposalOutput, nil
+	}
+
+	fmt.Println("send proposal in message: ", smsg.Cid())
```

Comment:
avoid fmt.Printlns in api side calls, if we really need to log something (which i dont think is necessary here) use `log.{Warn,Info,Debug}`